platform/api/aws/ec2.go

```diff
@@ -239,28 +251,246 @@ func (a *API) GetConsoleOutput(instanceID string) (string, error) {
 	sgIds, err := a.ec2.DescribeSecurityGroups(&ec2.DescribeSecurityGroupsInput{
-		GroupNames: []*string{&name},
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("group-name"),
+				Values: []*string{&name},
+			},
+		},
 	})
-	if isSecurityGroupNotExist(err) {
+
+	if len(sgIds.SecurityGroups) == 0 {
 		return a.createSecurityGroup(name)
 	}
 
 	if err != nil {
 		return "", fmt.Errorf("unable to get security group named %v: %v", name, err)
 	}
-	if len(sgIds.SecurityGroups) == 0 {
-		return "", fmt.Errorf("zero security groups matched name %v", name)
-	}
+
 	return *sgIds.SecurityGroups[0].GroupId, nil
 }
 
+// getVPCID gets a VPC for the given security group
+func (a *API) getVPCID(sgId string) (string, error) {
+	sgs, err := a.ec2.DescribeSecurityGroups(&ec2.DescribeSecurityGroupsInput{
+		GroupIds: []*string{&sgId},
+	})
+	if err != nil {
+		return "", fmt.Errorf("listing vpc's: %v", err)
+	}
+	for _, sg := range sgs.SecurityGroups {
+		if sg.VpcId != nil {
+			return *sg.VpcId, nil
+		}
+	}
+	return "", fmt.Errorf("no vpc found for security group %v", sgId)
+}
+
+// getSubnetID gets a subnet for the given VPC.
+func (a *API) getSubnetID(vpc string) (string, error) {
+	subIds, err := a.ec2.DescribeSubnets(&ec2.DescribeSubnetsInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("vpc-id"),
+				Values: []*string{&vpc},
+			},
+		},
+	})
+	if err != nil {
+		return "", fmt.Errorf("unable to get subnets for vpc %v: %v", vpc, err)
+	}
+	for _, id := range subIds.Subnets {
+		if id.SubnetId != nil {
+			return *id.SubnetId, nil
+		}
+	}
+	return "", fmt.Errorf("no subnets found for vpc %v", vpc)
+}
+
+// creates an InternetGateway and attaches it to the given VPC
+func (a *API) createInternetGateway(vpcId *string) (string, error) {
+	igw, err := a.ec2.CreateInternetGateway(&ec2.CreateInternetGatewayInput{})
+	if err != nil {
+		return "", err
+	}
+	if igw.InternetGateway == nil || igw.InternetGateway.InternetGatewayId == nil {
+		return "", fmt.Errorf("internet gateway was nil")
+	}
+	err = a.tagCreatedByMantle([]string{*igw.InternetGateway.InternetGatewayId})
+	if err != nil {
+		return "", err
+	}
+	_, err = a.ec2.AttachInternetGateway(&ec2.AttachInternetGatewayInput{
+		InternetGatewayId: igw.InternetGateway.InternetGatewayId,
+		VpcId:             vpcId,
+	})
+	if err != nil {
+		return "", fmt.Errorf("attaching internet gateway to vpc: %v", err)
+	}
+	return *igw.InternetGateway.InternetGatewayId, nil
+}
+
+// createSubnets creates a subnet in each availability zone for the region
+// that is associated with the given VPC associated with the given RouteTable
+func (a *API) createSubnets(vpcId, routeTableId *string) error {
+	azs, err := a.ec2.DescribeAvailabilityZones(&ec2.DescribeAvailabilityZonesInput{})
+	if err != nil {
+		return fmt.Errorf("retrieving availability zones: %v", err)
+	}
+
+	for i, az := range azs.AvailabilityZones {
+		// 15 is the maximum amount of zones possible when giving them a /20
```

Comment:
Returning on `i > 15` implies 16 zones.

Reply:
Ah, yeah, was referring to the variable rather than the actual amount. Updating the comment to be 16 & switching the error check to `i < 16` for consistency.